packages/standalone-uis/execution-timeline/src/app/components/change-detection-details/change-detection-details.component.ts

```diff
@@ -0,0 +1,71 @@
+import { Component, Input, OnChanges, OnDestroy, SimpleChanges } from '@angular/core';
+
+import { BridgeService } from '../../services/bridge.service';
+import { mapComponentTreeToFlameGraphTree } from '../../types/flame-graph-node/flame-graph-node.functions';
+import { FlameGraphNode } from '../../types/flame-graph-node/flame-graph-node.interface';
+import { round2 } from '../../util/misc-utils';
+
+function getComponentChangeDetections(componentInstances) {
+  const componentChangeDetections = new Map();
+  componentInstances.forEach((checkTime, instance) => {
+    if (!componentChangeDetections.has(instance.constructor)) {
+      componentChangeDetections.set(instance.constructor, { numChecks: 0 });
+    }
+
+    const entry = componentChangeDetections.get(instance.constructor);
+
+    entry.numChecks++;
+  });
+
+  return Array.from(componentChangeDetections.entries()).map(([componentType, entry]) => ({
+    componentType,
+    entry,
+  }));
+}
+
+@Component({
+  selector: 'ag-change-detection-details',
+  templateUrl: './change-detection-details.component.html',
+  styleUrls: ['./change-detection-details.component.scss'],
+})
+export class ChangeDetectionDetailsComponent implements OnChanges, OnDestroy {
+  @Input()
+  public segment: any;
+
+  public componentChangeDetections = null;
+  public rootFlameGraphNode: FlameGraphNode = null;
+  public runtimeInMilliseconds: number;
+
+  private subscription: any;
+
+  constructor(private bridge: BridgeService) {}
+
+  public ngOnChanges(changes: SimpleChanges): void {
+    this.subscription = this.bridge.subscribe(message => {
```

Comment:
Couldn't this cause a leak or unwanted side effects since it's on ngOnChanges?

Reply:
It could, but in this case it won't get called more than once - i verified.